### DISCUSSION TITLE
CO2 Sensor MH-Z19 implementation

### DISCUSSION THREAD

**Discussion by JohnnyB1989** [theyosh/TerrariumPI#779] [Q&A] [answered]
Hi all,

I have some problems with the MH-Z19 Sensor, or better how to get it to work in TerrariumPI.

I can read it via ssh, but what do I have to type in for the sensor address?

![Screenshot 2023-01-28 104238](https://user-images.githubusercontent.com/36489993/215259453-de9be1b4-6cc4-4cc3-a896-763d7881b714.png)

Thanks in advance.

**Comment by theyosh** [comment, marked as answer]
It should be something like `/dev/ttyS0` https://theyosh.github.io/TerrariumPI/hardware/sensor/mh-z19-co2-sensor/

**Reply by JohnnyB1989** [reply]
Hey Yosh,

already tried it, would not think about asking before reading the instructions :-)

I just get the response, the sensor could not be added.

**Reply by JohnnyB1989** [reply]
Okay... sorry to bother you, although it said its not possible to add the sensor, after two reboots, two CO2 sensors showed up :-)

**Reply by theyosh** [reply]
Did you try to add the sensors directly after a new install? Because after a first install a reboot is required. This CO2 sensor will not work if the Pi is not rebooted once after a new install.

**Reply by JohnnyB1989** [reply]
Could be :) but I´m not sure if I rebooted or not. Now its working, so thanks :-)
